docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+
+### Data relay
+
+IBC assumes the existence of a set of relayer processes with access to an underlying network protocol stack (likely TCP/IP, UDP/IP, or QUIC/IP) and physical interconnect infrastructure. These relayer processes monitor a set of ledgers implementing the IBC protocol, continuously scanning the state of each ledger and executing transactions on another ledger when outgoing datagrams have been committed. For correct operation and progress in a connection between two ledgers, IBC requires only that at least one correct and live relayer process exists which can relay between the ledgers.
+
+### Reliability
+
+The network layer and relayer processes may behave in arbitrary ways, dropping, reordering, or duplicating packets, purposely attempting to send invalid transactions, or otherwise acting Byzantine. This must not compromise the safety or liveness of IBC. This is achieved by assigning a sequence number to each packet sent over an IBC connection, which is checked by the IBC handler on the receiving ledger, and providing a method for the sending ledger to check that the receiving ledger has in fact received and handled a packet before sending more packets or taking further action. Cryptographic commitments are used to prevent datagram forgery: the sending ledger commits to outgoing datagrams, and the receiving ledger checks these commitments, so datagrams altered in transit by a relayer will be rejected.
+
+### Flow control
+
+IBC does not require specific provision for computation-level flow control since the underlying ledgers will have throughput limitations and flow control mechanisms of their own (such as "gas" markets). Application-level flow control — limiting the rate of particular packets according to their content — may be useful to ensure security properties (limiting the value on a single ledger) and contain damage from Byzantine faults (allowing a challenge period to prove an equivocation, then closing a connection). IBC provides facilities for modules to reject packets and leaves particulars up to the higher-level application protocols.
+
+### Authentication
+
+All datagrams in IBC are authenticated: a block finalized by the consensus algorithm of the sending blockchain must commit to the outgoing datagram via a cryptographic accumulator, and the receiving chain must verify both the consensus transcript and the accumulator proof that the datagram was sent (and associated actions executed) before acting upon it. For correct execution semantics, both chains must implement the correct IBC protocol — of course, since IBC is permissionless, any chain could connect to another chain and provide a correct accumulator proof but incorrectly implement IBC such that its IBC handler did not behave according to protocol, so modules utilizing IBC connections (or end-users using those modules) must inspect the state of the connections and ledgers they plan to use to ensure correct implementation.
```

Comment:
Blockchain and ledger are being used interchangeably. If these two terms are assuming the same thing, maybe it make sense to state this in vocabulary section.

Reply:
This paragraph is not at the same level of abstraction/details as previous part of the document. Can we explain what is meant by "communication is authenticated" without going too much into details?The sentence about IBC handler not behaving according to the protocol is confusing. It seems that there is an implicit assumptions that IBC guarantees are guaranteed to hold only if IBC modules are correct?This might require some clarification. Do we assume the same fault model for IBC module and application module that are collocated?

Reply:
I think we can get rid of that sentence for now, I agree it's too low-level. Done.

Clarified blockchain / ledger at the top of the document.